sdk/storage/common/storage_checksum.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <string>
+
+namespace azure { namespace storage { namespace common {
+
+enum checksum_type
+{
+    none,
+    md5,
+    sha256,
+    crc64
+};
+
+class checksum
+{
+public:
+    checksum(checksum_type type, std::string value) : m_type(type), m_value(std::move(value)) {}
+    checksum(checksum_type type, uint64_t value)
+    {
+        if (type != checksum_type::crc64)
+        {
+            throw storage_exception("Invalid checksum type");
+        }
+        m_type = type;
+        // TODO: convert int value to string notation
+    }
+
+    static checksum none()
```

Comment:
What if it returns a const ref of c, where c is initialized with `static checksum c = checksum(checksum_type::none, std::string());`.

Reply:
okay, that will also work.